crates/vm/backends/revm/mod.rs

```diff
@@ -624,7 +629,7 @@ pub fn extract_all_requests(
     header: &BlockHeader,
 ) -> Result<Vec<Requests>, EvmError> {
     let config = state.inner.database.get_chain_config()?;
-    let spec_id = spec_id(&config, header.timestamp);
+    let spec_id = spec_id(&config, header.timestamp, header.difficulty);
```

Comment:
This is the same discussion as https://github.com/lambdaclass/ethrex/pull/3111#discussion_r2140126514 but with a different field :/